ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -362,30 +362,30 @@ static char *pdo_pgsql_last_insert_id(pdo_dbh_t *dbh, const char *name, size_t *
 	pdo_pgsql_db_handle *H = (pdo_pgsql_db_handle *)dbh->driver_data;
 	char *id = NULL;
 
-	if (name == NULL) {
-		if (H->pgoid == InvalidOid) {
-			return NULL;
-		}
-		*len = spprintf(&id, 0, ZEND_LONG_FMT, (zend_long) H->pgoid);
+	PGresult *res;
+	ExecStatusType status;
+	const char *q[1];
+	q[0] = name;
+
+	if (PHP_PDO_PGSQL_LASTVAL_PG_VERSION <= PQserverVersion && name == NULL) {
```

Comment:
PQserverVersion is a function, you currently compare teh function pointer.

Thanks.